File: c++/p004.cpp

#include <string>
#include <algorithm>
#include <list>
#include <iostream>

using namespace std;

bool isPalendromic(int n) {
	string s = to_string(n);
	string c = to_string(n);
	reverse(c.begin(), c.end());
	return c == s;
}

int main() {
	list<int> palendromes = {};
	for (int i = 100; i < 1000; i++) {
		for (int j = 100; j < 1000; j++) {
			int prod = i * j;
			if( isPalendromic(prod) ) {
				palendromes.push_back(prod);
			}
		}
	}
	palendromes.sort();
	int parr[palendromes.size()];
	copy(palendromes.begin(), palendromes.end(), parr);
	return parr[palendromes.size()-1];
}

File: php/p004.php
function isPalendromic($n) {
	$s = strval($n);
	$c = strrev($s);
	return $s == $c;
}

function compute() {
	$palendromes = array();
	for ($i = 100; $i < 1000; $i++) {
		for ($j = 100; $j < 1000; $j++) {
			$prod = $j * $i;
			if (isPalendromic($prod)) {
				array_push($palendromes, $prod);
			}
		}
	}
	return max($palendromes);
}
echo compute();